src/steps/analyze-project/analyze-components/find-arguments.ts

```diff
@@ -1,8 +1,25 @@
 // import { AST as ASTJavaScript } from '@codemod-utils/ast-javascript';
-// import { AST as ASTTemplate } from '@codemod-utils/ast-template';
+import { AST as ASTTemplate } from '@codemod-utils/ast-template';
 
 import type { Signature } from '../../../types/index.js';
 
+function analyzeTemplate(file: string, args: Set<string>): void {
+  const traverse = ASTTemplate.traverse();
+
+  traverse(file, {
+    PathExpression(node) {
+      if (node.head.type !== 'AtHead') {
+        return;
+      }
+
+      const argumentName = node.original.replace(/^@/, '');
+      const key = argumentName.split('.')[0]!;
+
+      args.add(key);
```

Comment:
If an argument is an object (a POJO), `argumentName` (on line 15) will look something like `product.name` and `product.imageUrl`.

For simplicity, we won't try to reconstruct the object's shape, but always denote the argument's type as `unknown`. This will also help us ignore the case where the actual type needs to be imported from some other file.